formencode/validators.py

```diff
@@ -1351,6 +1364,10 @@ def _validate_python(self, value, state):
             raise Invalid(
                 self.message('badUsername', state, username=username),
                 value, state)
+        if hasattr(self, 'domain_blacklist') and domain in self.domain_blacklist:
```

Comment:
Domain names are case insensitive but there is no handling of that here. You could have ```mybaddomainname.com``` in ```domain_blacklist``` and the user could get by this check with ```haha@MyBadDomainName.com```. There should probably be some case normalization of ```domain_blacklist``` also.